airflow/providers/amazon/provider.yaml

```
@@ -67,6 +66,7 @@ dependencies:
   - mypy-boto3-rds>=1.24.0
   - mypy-boto3-redshift-data>=1.24.0
   - mypy-boto3-appflow>=1.24.0
+  - aiobotocore[boto3]
```

```suggestion
  - aiobotocore[boto3]>=2.2.0
```
We had `boto3>=1.24.0` for a reason
we should make sure that we don't allow regression of boto3 version
https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst#220-2022-03-16

---

The boto3 version comes out to be 1.26.76, with the constraints I had.
```
root@fba2b84bfa3f:/opt/airflow# pip show aiobotocore
Name: aiobotocore
Version: 2.5.0
Summary: Async client for aws services using botocore and aiohttp
Home-page: https://github.com/aio-libs/aiobotocore
Author: Nikolay Novik
Author-email: nickolainovik@gmail.com
License: Apache License 2.0
Location: /usr/local/lib/python3.7/site-packages
Requires: aiohttp, aioitertools, botocore, wrapt
Required-by: 
root@fba2b84bfa3f:/opt/airflow# pip show botocore
Name: botocore
Version: 1.29.76
Summary: Low-level, data-driven core of boto 3.
Home-page: https://github.com/boto/botocore
Author: Amazon Web Services
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.7/site-packages
Requires: jmespath, python-dateutil, urllib3
Required-by: aiobotocore, aws-xray-sdk, boto3, moto, redshift-connector, s3transfer
root@fba2b84bfa3f:/opt/airflow# pip show boto3
Name: boto3
Version: 1.26.76
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.7/site-packages
Requires: botocore, jmespath, s3transfer
Required-by: aws-sam-translator, moto, redshift-connector, watchtower
root@fba2b84bfa3f:/opt/airflow# 
```

---

Users may install the providers separately with their own constraint mechanism which depends on users list of requested packages may result in using older versions.
PIP always tries to have the latest version possible but if it can't then it start backtrack to older versions. By setting lower limit we instruct PIP not to look below this version.